### PR TITLE
chore(sync): blockchain tree config and externals

### PR DESCRIPTION
**Pull request by rkrasiuk**
Extract blockchain tree config and externals

### CODE REVIEW

File: crates/executor/src/blockchain_tree/config.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    /// at least `additional_canonical_block_hashes`+`max_reorg_depth`, for eth that would be
+    /// 256+64.
+    num_of_additional_canonical_block_hashes: u64,
+}
+
+impl Default for BlockchainTreeConfig {
+    fn default() -> Self {
+        Self {
+            max_reorg_depth: 64,
+            max_blocks_in_chain: 0, // TODO:
```
Comment (rkrasiuk) [inline crates/executor/src/blockchain_tree/config.rs:21]: @rakita what's the reasonable default here?

File: crates/executor/src/blockchain_tree/factory.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+impl<DB, C, EF> BlockchainTreeFactory<DB, C, EF>
+where
+    DB: Database + Clone,
+    C: Consensus + Clone,
+    EF: ExecutorFactory + Clone,
+{
+    /// Create instance of the blockchain tree.
+    pub fn create_tree(&self) -> Result<BlockchainTree<DB, C, EF>, Error> {
```
Comment (rakita) [inline crates/executor/src/blockchain_tree/factory.rs:80]: `create_tree` seems strange wih cloning of externals, I would expect builder here, and to have `fn build` that consumes structure and returns BlockchainTree and would call it `BlockchainTreeBuilder`

File: crates/executor/src/blockchain_tree/block_indices.rs [hunk 3]
```diff
@@ -258,6 +246,7 @@ impl BlockIndices {
     pub fn finalize_canonical_blocks(
         &mut self,
         finalized_block: BlockNumber,
+        num_blocks_to_retain: u64,
```
Comment (rakita) [inline crates/executor/src/blockchain_tree/block_indices.rs:249]: Can you use the full name here `num_of_additional_canonical_hashes_to_ratain`

**Review by rakita — APPROVED** [review]
lgtm

File: crates/executor/src/blockchain_tree/config.rs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    /// 256+64.
+    num_of_additional_canonical_block_hashes: u64,
+}
+
+impl Default for BlockchainTreeConfig {
+    fn default() -> Self {
+        Self {
+            max_reorg_depth: 64,
+            max_blocks_in_chain: 65,
+            num_of_additional_canonical_block_hashes: 256,
```
Comment (mattsse) [inline crates/executor/src/blockchain_tree/config.rs:22]: can we add a note about how we came up with these or that they're just assumptions

**Review by gakonst — APPROVED** [review]
@rkrasiuk nice work - FYI we'll have a bunch of changer in these as #1674 refactors a lot of that code

**Pull request merged**